proof-systems/src/groth16/benches/gro16_bench.rs

```diff
@@ -0,0 +1,213 @@
+#[macro_use]
+extern crate criterion;
+
+use algebra::{
+    fields::bn_382::Fr,
+    curves::bn_382::Bn382,
+    UniformRand, PrimeField, Field,
+};
+use r1cs_core::{SynthesisError, ConstraintSynthesizer, ConstraintSystem, LinearCombination};
+use proof_systems::groth16::{generate_random_parameters, create_random_proof};
+
+use rand::SeedableRng;
+use rand_xorshift::XorShiftRng;
+
+use criterion::{BenchmarkId, BatchSize};
+use criterion::Criterion;
+
+use std::marker::PhantomData;
+
+pub struct Benchmark<F: PrimeField> {
+    inputs: Vec<Option<F>>,
+    num_constraints: usize,
+}
+
+impl<F: PrimeField> ConstraintSynthesizer<F> for Benchmark<F> {
```

Comment:
Let's add a comment on the specifics of this test circuit.

Reply:
Done